src/main/java/Deadline.java

```diff
@@ -0,0 +1,28 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private String byWhen;
+    private LocalDate deadline;
+
+    public Deadline(String item, boolean marked, String byWhen) {
+        super(item, marked);
+        this.byWhen = byWhen;
+        this.deadline = LocalDate.parse(byWhen);
+    }
+
+    public String getbyWhen() {
+        return this.byWhen;
+    }
+
+    @Override
+    public String stringify() {
+        String m = "";
+        if (super.isMarked())
+            m = "[X]";
+        else
+            m = "[ ]";
+        return "[D]" + m + " " + super.stringify() + " (by: "
+                + this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
```

Comment:
It's not necessary to use the keyword `super` here, and I think it makes the code harder to read. On the other hand, `super.stringify()` is definitely an appropriate usecase for it so good job for that👍. Basically, you should only use `super` when you need the superclass's implementation of the method instead of the one in your current class.
```suggestion
        if (isMarked())
            m = "[X]";
        else
            m = "[ ]";
        return "[D]" + m + " " + super.stringify() + " (by: "
                + this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
    }
```